lib/ansible/modules/packaging/os/pacman.py

```diff
@@ -191,18 +191,26 @@ def upgrade(module, pacman_path):
     cmdupgrade = "%s -Suq --noconfirm" % (pacman_path)
     cmdneedrefresh = "%s -Qqu" % (pacman_path)
     rc, stdout, stderr = module.run_command(cmdneedrefresh, check_rc=False)
+    data = stdout.split('\n')
+    data.remove('')
+    diff = {
+        'before': '',
+        'after': '',
+        'after_header': 'upgraded'
+    }
 
     if rc == 0:
+        if module._diff:
+            diff['after'] = '\n'.join(data) + '\n'
         if module.check_mode:
-            data = stdout.split('\n')
-            module.exit_json(changed=True, msg="%s package(s) would be upgraded" % (len(data) - 1))
+            module.exit_json(changed=True, msg="%s package(s) would be upgraded" % (len(data)), packages=data, diff=diff)
         rc, stdout, stderr = module.run_command(cmdupgrade, check_rc=False)
         if rc == 0:
-            module.exit_json(changed=True, msg='System upgraded')
+            module.exit_json(changed=True, msg='System upgraded', packages=data, diff=diff)
         else:
             module.fail_json(msg="Could not upgrade")
     else:
-        module.exit_json(changed=False, msg='Nothing to upgrade')
+        module.exit_json(changed=False, msg='Nothing to upgrade', packages=data)
```

Comment:
Any non-standard exit keywords should be [documented](http://docs.ansible.com/ansible/dev_guide/developing_modules.html#example) in the `RETURN` block.